### PR TITLE
Simplifies input equation to remove brackets and GUI bug fix

### PR DESCRIPTION
**Pull request by abhishekti7**
Earlier when equation contained multiple nested brackets, the program would display "Invalid Expression".
This commit adds a function to preprocess the input equation to simplify any such nested bracket situation.

commit

### CODE REVIEW

**Comment by abhishekti7** [comment]
#129

File: tests/test_simplify.py [hunk 1]
```diff
@@ -40,6 +41,12 @@ def test_simplify():
     assert quickTest("x/y + x/x + x/x^2 + x^2/x = x/y^2 + x^2/y + x - 1", simplifyEquation) == "xy^(-1)+2.0+x^(-1.0)-xy^(-2.0)-x^(2.0)y^(-1)=0"
 
 
+def test_preprocess_simplification():
+
+    assert preprocessSimplification("4+(4+(4+1))") == "4+4+4+1"
+    assert preprocessSimplification("4-(4+(4-1))") == "4-4-4+1"
```
Comment (8hantanu) [inline tests/test_simplify.py:47]: Input cases with * and / might not return the correct result.

Comment (abhishekti7) [reply tests/test_simplify.py:47]: yeah I checked that and that's why I left a note stating that this pr does not simplify the input equation. It is just a temporary fix for the input case that was in the issue referenced here.

Comment (8hantanu) [reply tests/test_simplify.py:47]: I am talking about a case like `4-(4*(4-1))` which will reduce to `4-4*4+1`.

Comment (abhishekti7) [reply tests/test_simplify.py:47]: Okay I am looking into this. Thank you.

**Comment by abhishekti7** [comment]
#144 ~Since by definition, matplotlib.pyplot.suptitle creates a centered headline therefore there will always be a chance that the rendered latex overflows unless we set the **horizontalalignment** attribute to *left*~ In the meantime the issue can be fixed by stating the default x-coordinate.
lol

**Pull request closed without merging**